src/Workspaces/Core/Portable/Workspace/Solution/ProjectState_Checksum.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Serialization;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial class ProjectState
+    {
+        public bool TryGetStateChecksums(out ProjectStateChecksums stateChecksums)
+        {
+            return _lazyChecksums.TryGetValue(out stateChecksums);
+        }
+
+        public async Task<Checksum> GetChecksumAsync(CancellationToken cancellationToken)
```

Comment:
> not sure why we made GetText can't be called on any thread when compiler is free to do so.

Because the IDE layer avoids synchronous work due to the blocking impact it has on threads.  We don't want IO ending up starving threads.    The compiler layer is different as we try to ensure that hte data it needs is warm and local (so it won't actually block threads on IO when makign sync calls).  It's not a perfect mesh, but it fits the different design goals of each layer.